src/_nebari/template/stages/07-kubernetes-services/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -311,6 +311,61 @@ def configure_user(username, groups, uid=1000, gid=100):
     }
 
 
+def profile_argo_token(groups):
+    # TODO: create a more robust check user's Argo-Workflow role
+
+    domain = z2jh.get_config("custom.external-url")
+
+    ADMIN = "admin"
+    DEVELOPER = "developer"
+    ANALYST = "analyst"
+
+    base = "argo-"
+
+    if ANALYST in groups:
+        argo_sa = base + "view"
+    if DEVELOPER in groups:
+        argo_sa = base + "dev"
+    if ADMIN in groups:
+        argo_sa = base + "admin"
+    else:
+        return {}
+
+    return {
+        "ARGO_BASE_HREF": "/argo",
+        "ARGO_SERVER": f"{domain}:443",
+        "ARGO_TOKEN": {
+            "valueFrom": {
+                "secretKeyRef": {
+                    "name": f"{argo_sa}.service-account-token",
+                    "key": "token",
+                }
+            }
+        },
+    }
+
+
+def profile_conda_store_viewer_token():
+    return {
+        "CONDA_STORE_TOKEN": {
```

Comment:
Any way this token could be misused by the user?  It's used by ARGO_WORKFLOWS_EXECUTOR, I see - https://github.com/search?q=repo%3Anebari-dev%2Fargo-workflows-executor+CONDA_STORE_TOKEN&type=code

Reply:
Is Argo Workflows Executor a package that's installable or a pod that's running or its included on the jupyterhub pod?

Reply:
This token is view-only in scope so if the user was motivated enough, they could view all of the namespaces/environments that exist.

Reply:
Later down the road we should scope this token to just what the user can see. But for now this works.